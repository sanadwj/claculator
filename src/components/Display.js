import React from 'react';
import PropTypes from 'prop-types';
import '../styles/App.css';

const Display = props => {
  const { total, operation, next } = props;
  return (
    <h1 className="display">
      {(!total && next) || total }
      {operation }
      {(total && next)}
    </h1>
  );
};

Display.propTypes = {
  total: PropTypes.string,
  operation: PropTypes.string,
  next: PropTypes.string,
};

Display.defaultProps = {
  total: String,
  operation: String,
  next: String,
};

export default Display;
