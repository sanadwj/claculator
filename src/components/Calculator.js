import React, { useState } from 'react';
import Display from './Display';
import ButtonPanel from './ButtonPanel';
import calculate from '../logic/calculate';
import '../styles/App.css';

const Calculator = () => {
  const [state, setState] = useState({
    total: '',
    operation: '',
    next: '',
  });

  const handleClick = buttonName => {
    const result = calculate(state, buttonName);
    setState(result);
  };

  return (
    <>
      <div className="calculate">
        <h2>Let`s do some math!</h2>
        <div className="calculator">
          <Display total={state.total} operation={state.operation} next={state.next} />
          <ButtonPanel clickHandler={handleClick} />
        </div>
      </div>
    </>

  );
};

export default Calculator;
