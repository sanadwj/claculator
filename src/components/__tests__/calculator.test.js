import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import React from 'react';
import renderer from 'react-test-renderer';

import Calculator from '../Calculator';
import Display from '../Display';

describe('Calculator', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <Calculator />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Creates component', () => {
    render(<Calculator><Display total="9" /></Calculator>);
    const element = screen.getByText('9');
    expect(element).toBeInTheDocument();
  });
});
