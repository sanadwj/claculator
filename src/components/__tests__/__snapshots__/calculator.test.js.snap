// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator renders correctly 1`] = `
<div
  className="calculate"
>
  <h2>
    Let\`s do some math!
  </h2>
  <div
    className="calculator"
  >
    <h1
      className="display"
    >
      
      
      
    </h1>
    <div
      className="buttonPanel"
    >
      <div
        className="buttonDiv"
      >
        <button
          className="sc-bdfBwQ cwcmQO button"
          name="AC"
          onClick={[Function]}
          type="button"
        >
          <h3>
            AC
          </h3>
        </button>
        <button
          className="sc-gsTCUz gSTbTe button"
          name="+/-"
          onClick={[Function]}
          type="button"
        >
          <h3>
            +/-
          </h3>
        </button>
        <button
          className="sc-dlfnbm gCdEnx button"
          name="%"
          onClick={[Function]}
          type="button"
        >
          <h3>
            %
          </h3>
        </button>
        <button
          className="sc-hKgILt jmFaqI button"
          name="รท"
          onClick={[Function]}
          type="button"
        >
          <h3>
            รท
          </h3>
        </button>
      </div>
      <div>
        <button
          className="sc-eCssSg kMFudk button"
          name="7"
          onClick={[Function]}
          type="button"
        >
          <h3>
            7
          </h3>
        </button>
        <button
          className="sc-jSgupP hywpap button"
          name="8"
          onClick={[Function]}
          type="button"
        >
          <h3>
            8
          </h3>
        </button>
        <button
          className="sc-gKsewC jlpHw button"
          name="9"
          onClick={[Function]}
          type="button"
        >
          <h3>
            9
          </h3>
        </button>
        <button
          className="sc-iBPRYJ kAZABr button"
          name="X"
          onClick={[Function]}
          type="button"
        >
          <h3>
            X
          </h3>
        </button>
      </div>
      <div>
        <button
          className="sc-fubCfw kDLxcF button"
          name="4"
          onClick={[Function]}
          type="button"
        >
          <h3>
            4
          </h3>
        </button>
        <button
          className="sc-pFZIQ lcfLyW button"
          name="5"
          onClick={[Function]}
          type="button"
        >
          <h3>
            5
          </h3>
        </button>
        <button
          className="sc-jrAGrp eDvsNC button"
          name="6"
          onClick={[Function]}
          type="button"
        >
          <h3>
            6
          </h3>
        </button>
        <button
          className="sc-kEjbxe kZkilW button"
          name="-"
          onClick={[Function]}
          type="button"
        >
          <h3>
            -
          </h3>
        </button>
      </div>
      <div>
        <button
          className="sc-iqHYGH fuiasQ button"
          name="1"
          onClick={[Function]}
          type="button"
        >
          <h3>
            1
          </h3>
        </button>
        <button
          className="sc-crrsfI hZejzN button"
          name="2"
          onClick={[Function]}
          type="button"
        >
          <h3>
            2
          </h3>
        </button>
        <button
          className="sc-dQppl kRihZH button"
          name="3"
          onClick={[Function]}
          type="button"
        >
          <h3>
            3
          </h3>
        </button>
        <button
          className="sc-bqyKva jrQojn button"
          name="+"
          onClick={[Function]}
          type="button"
        >
          <h3>
            +
          </h3>
        </button>
      </div>
      <div>
        <button
          className="sc-kstrdz eYQnEU button"
          name="0"
          onClick={[Function]}
          type="button"
        >
          <h3>
            0
          </h3>
        </button>
        <button
          className="sc-hBEYos dMuZIM button"
          name="."
          onClick={[Function]}
          type="button"
        >
          <h3>
            .
          </h3>
        </button>
        <button
          className="sc-fodVxV kvMFpM button"
          name="="
          onClick={[Function]}
          type="button"
        >
          <h3>
            =
          </h3>
        </button>
      </div>
    </div>
  </div>
</div>
`;
